Packs/Datadog/Integrations/Datadog/Datadog.py

```diff
@@ -0,0 +1,292 @@
+    else:
+        PARAMS = None
+    res = http_request('GET', '/api/v2/users', params=PARAMS)
+    if res.get('data'):
+        userList = []
+        userDetails = {}
+        for items in res['data']:
+            userDetails = flattenRow(items['attributes'])
+            userDetails['id'] = items['id']
+            userDetails['type'] = items['type']
+            userList.append(userDetails)
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'EntryContext': {'dataDog.users': userList},
+            'Contents': userList,
+            'HumanReadable': tableToMarkdown("Users", userList, ["id", "name", "email", "status"])})
+    else:
+        demisto.results("No users returned")
+
+
+def list_active_metrics():
+    PARAMS = demisto.args()
+    res = http_request('GET', '/api/v1/metrics', params=PARAMS)
+    if res.get('metrics'):
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['json'],
+            'Contents': res['metrics'],
+            'HumanReadable': tableToMarkdown("Metrics", res['metrics'], ["Metrics"])})
+    else:
+        demisto.results("No metrics returned")
+
+
+def submit_metics():
+    data = {
+        "series": [{
+            "host": str(demisto.args()['host']),
+            "interval": int(demisto.args()['interval']),
+            "type": str(demisto.args()['type']),
+            "metric": str(demisto.args()['metric']),
+            "points": [[str(demisto.args()['pointTimeStamp']), str(demisto.args()['pointValue'])]]
+        }]
+    }
+    res = http_request('POST', '/api/v1/series', DATA=json.dumps(data))
+    if res.get("status"):
+        if res['status'] == "ok":
+            demisto.results("Metrics updated successfully")
+        else:
+            demisto.results("Metrics update failed")
+
+
+def fetch_incidents():
+    now = round(time.time())
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('last_fetch', None)
+    if last_fetch is None:
+        # if this is the first run, fetch events for the last 3 hours
+        startTime = now - 10800
+    else:
+        startTime = last_fetch
+    latest_created_time = startTime
+    PARAMS = {
+        "start": str(startTime),
+        "end": str(now),
+        "priority": demisto.params()['incidentPriority'],
+        "unaggregated": True
+    }
+    res = http_request('GET', '/api/v1/events', params=PARAMS)
+    if res.get('events'):
+        incidents = []
+        for event in res['events']:
+            incident_created_time = int(event.get('date_happened', '0'))
+            incident_created_time_ms = incident_created_time * 1000
+            incident = {
+                "name": event['title'],
+                "occurred": timestamp_to_datestring(incident_created_time_ms),
+                "rawJSON": json.dumps(event)
+            }
+            if incident_created_time > latest_created_time:
```

Comment:
Can it be that two incidents happened at the same time (perhaps with a millisecond gap, which is not covered), and only one was fetched in the previous round? If so, in this round the second incident will not be fetched.
We must not miss events, so if this might happen we should find a solution.